src/rust/runtime/memory/demibuffer.rs

```diff
@@ -747,12 +747,7 @@ impl DemiBuffer {
     /// long as there is a DemiBuffer in existance that is holding a reference on this data.
     // This function is not marked unsafe, as the unsafe act is dereferencing the returned pointer, not providing it.
     pub fn as_ptr(&self) -> *const u8 {
-        // TODO: Review having this "match", since MetaData and MBuf are laid out the same, these are equivalent cases.
-        match self.get_tag() {
-            Tag::Heap => self.data_ptr(),
-            #[cfg(feature = "libdpdk")]
-            Tag::Dpdk => self.dpdk_data_ptr(),
-        }
+        self.data_ptr()
     }
```

Comment:
Since we no longer need the multiplexing logic, we should also move the implementation of `data_ptr()` to `as_ptr()` and get rid of `data_ptr()`.

Reply:
sounds good!